relay_messages.go

```diff
@@ -206,6 +217,20 @@ func (f lazyCallReq) Arg2StartOffset() int {
 	return f.arg2StartOffset
 }
 
+// Arg2Iterator returns the iterator for reading Arg2 key value pair
+// of TChannel-Thrift Arg Scheme.
+func (f lazyCallReq) Arg2Iterator() (arg2.KeyValIterator, error) {
+	if !f.hasTChanThrift {
+		return arg2.KeyValIterator{}, io.EOF
```

Comment:
if we have calls that are not tchannel, can we add more context on what we got? maybe a different error, and should we try to capture the actual arg scheme we saw?

Reply:
Updated - was thinking to save some info, but I guess bool won't be too far from []byte anyway.